### DISCUSSION TITLE
How to change the default shell to zsh ?

### DISCUSSION THREAD

**Discussion by xd003** [git-for-windows/git#4617] [Q&A]
Hello, i would like to change the default shell from bash to zsh. I couldn't find much info on this. At the moment i have added the following to the bottom of ```"C:\Program Files\Git\etc\bash.bashrc"``` to make bash start zsh for me.
```
if [ -t 1 ]; then
  exec zsh
fi
```

This works but i would like to know a way to load zsh directly by default, that would improve the prompt startup time on my low end machine as it wouldn't have to load multiple shells

MSYS2 has option to change default shell in a very easy/graceful manner as follows :
We can either set the login shell of everyone using our MSYS2 installation using [nsswitch.conf](https://cygwin.com/cygwin-ug-net/ntsec.html#ntsec-mapping-nsswitch) or use [mkpasswd -c](https://cygwin.com/cygwin-ug-net/mkpasswd.html) to create an /etc/passwd entry that we can modify.

Since git bash is based on MSYS2, i expected this to work out but adding ```db_shell: /usr/bin/zsh``` in nsswitch.conf inside the /etc directory seems to have no effect unfortunately. Alternatively running ```mkpasswd -c``` doesn't seem to create any ```/etc/passwd``` entry

**Comment by rimrul** [comment]
This is definitely unsupported. What you did is probably the easiest way to do it, but I don't know where you even got a `zsh` from.

> This works but i would like to know a way to load zsh directly by default

I guess 

https://github.com/git-for-windows/git/wiki/Install-inside-MSYS2-proper

and then passing `-shell zsh` to `msys2_shell.cmd` could work.

**Comment by xd003** [comment]
>  but I don't know where you even got a `zsh` from.

There are 2 ways to get zsh on git bash 
1) Setup pacman properly and then just install zsh through that ```sudo pacman -s zsh```
2) or if you don't want pacman, just manually download it from https://repo.msys2.org/msys/x86_64/zsh-doc-5.9-2-x86_64.pkg.tar.zst and you need to move the relevant directories in your git bash installtion,  i can post the specific instruction later if you need it
